orchestra/actions/action.py

```diff
@@ -118,6 +118,9 @@ def _target_name(self):
 
     @property
     def source_dir(self) -> str:
+        if self.component.clone:
+            return os.path.join(self.config.sources_dir, self.component.clone.repository)
+
```

Comment:
Shouldn't `source_dir` fail if we don't have `component.clone`?

Reply:
IMO no, the env var `SOURCE_DIR` could also be used by components without a repo (and it used to be for some components getting sources from tarballs that did an out-of-tree build, not sure it it's still that way)

Reply:
Mmmh, "getting sources from tarballs" is something that's just part of the configure script. orchestra has no idea about it.

I think we should fail in absence of `component.clone`.

Reply:
I tried to fail in case there's no `component.clone`, but there's quite a few places that assume `SOURCE_DIR` exists.
I fixed some of them, but there's more.
One day we'll need to fix this.